### ISSUE TITLE
Large file support in Postman

### ISSUE THREAD

**Issue by a85** [a85/POSTMan-Chrome-Extension#1946]
Postman has problems in supporting large files. It was not designed for heavy loads but we intend to take this up as a priority now. Adding all issues from Github below and close them so that we have one place to track this:

https://github.com/postmanlabs/postman-app-support/issues/555
https://github.com/postmanlabs/postman-app-support/issues/791
https://github.com/postmanlabs/postman-app-support/issues/1188
https://github.com/postmanlabs/postman-app-support/issues/1117
https://github.com/postmanlabs/postman-app-support/issues/1089
https://github.com/postmanlabs/postman-app-support/issues/1301
https://github.com/postmanlabs/postman-app-support/issues/1843
https://github.com/postmanlabs/postman-app-support/issues/2112

**Comment by shamasis** [comment]
Status:

Request:
- form-data mode: fixed
- binary mode: wip

Response: underlying streams ready. need to tweak features to not consume response as a string, instead use Buffer. That may affect a number of features around preview and other such features.

**Comment by shamasis** [comment]
Binary mode should now play well - Mac app v4.5.1+

**Comment by Celadora** [comment]
4.10.2 (Windows 7) crashes whenever we try to upload a file of about 420mb through form-data

**Comment by dangle1** [comment]
v4.10.7 (Windows 7) crashes for me when receiving a 1.89 mb JSON response as well.

**Comment by dsomel21** [comment]
@a85 didn't you create Postman?

**Comment by skoold2003** [comment]
V5.1.1 Mac crashes with 14.1MB JSON response. Kind of an extreme example but I don't have a payload to test that's smaller. Works in Safari on the same query.

**Comment by Jasonlhy** [comment]
Using raw response will be much better if the response size is over some threshold.

**Comment by mikecg** [comment]
Is this supposed to be fixed? I cannot retrieve a 40mb JSON response payload. Using Windows v5.5.0. Postman eventually hangs and I need to kill the process. The request works in Chrome, as well as in Insomnia.

**Comment by brazenvoid** [comment]
Yes, it hangs still (on windows). I have 897KB file which comes as part of the JSON response and it hangs like no tomorrow.

**Comment by dangithubs** [comment]
I think the issue must still exist as I am getting the same behaviour. It displays some of the data retrieved from the endpoint and then the postman panel just goes white and locks up. You then have to manually kill the process. I'm running Version 4.10.2 as the updater seems to be broken as well so it wont update to 5.5.0.

**Comment by HeliosLHD** [comment]
I would have to concur.  Version 5.5.2 Windows 10 64-bit.  I receive a 3.38MB JSON response and it shows response time as 1735ms, but takes nearly a minute to process before showing the text in the response window, (Pretty/Raw/Preview) doesn't matter.

I can provide the cURL call to retrieve this in PM, just can't post to open forum.

Watching Task Manager, even just swapping between Raw<->Pretty causes a spike in CPU and memory usage, but it eventually returns within 30-60 seconds.
![image](https://user-images.githubusercontent.com/13319992/35802881-cd5ced46-0a3f-11e8-9ab7-e32a95421132.png)

**Comment by vinogradoff** [comment]
@HeliosLHD same here. It takes up to 10 secs to process 1Mb response (response time itself is 200ms), and it never ends (30 mins still waiting...) with 4Mb response.

**Comment by brazenvoid** [comment]
I think that there should be an option to just hide the file content and not process it, a manually settable length for such would be best. 

One may just show a message where the file/data is supposed to be with a small preview and then the total length.

**Comment by ddfreiling** [comment]
I vote for a smaller preview option or one to entirely hide it, for very large responses. Especially for binary responses.

**Comment by AmyGao18** [comment]
Running postman in latest version v6.0.10, still getting same issue with a large response body,  in a few minutes postman crashes. :-(

**Comment by rohit0903** [comment]
Do we have any updates ?? Facing the same issue with 300+MB size zip file

**Comment by AmyGao18** [comment]
Nope, ☹

**Comment by pnatraj** [comment]
The response was of 176MB and postman app on Mac kept crashing...

du -sh something.log 176M	


[postmanbug.txt](https://github.com/postmanlabs/postman-app-support/files/2201247/postmanbug.txt)

**Comment by lucamaf** [comment]
version 6.7.2 on Fedora crashes with POST and binary/stream format of 300MB

**Comment by i3anaan** [comment]
Experiencing crashes when trying to use the runner for testing performance on a 4.3Mb js file.

**Comment by nicholemattera** [comment]
Just experienced a crash on Linux Mint 19 for a 9.6MB file. This should be a pretty simple fix if the request comes back with a Content-Disposition header as an attachment then it shouldn't display the data as raw, and should probably just have a download button.

**Comment by eighteyes** [comment]
+1 trying to use POSTMAN to test an API and it crashes when downloading a file.

**Comment by neawasthi** [comment]
Postman keep crashing when downloading big response (~120MB) and is unusable. 
I have used Fiddler in past which handles this intelligently. It truncates the display within the tool and allows user to load the file outside of the application (in Vim etc)
Please fix or let us know how to work with large files..

**Comment by carlmwright** [comment]
![postman](https://user-images.githubusercontent.com/26304283/56819001-c8594500-6840-11e9-849a-6fc81884affb.png)

This is about where my Postman 7.0.9 x64 decides to call it a day when downloading an ~80Mb application/xml gzipped to ~5Mb.

Running fiddler over it shows that it all comes back to the client in about 27 seconds, so maybe it's some sort of pretty printing that it's doing?

**Comment by SamPlusPlus** [comment]
My team often deals with downloading Json anywhere from 10mb-100mb it appears Postman cannot handle this at all for us. It often just hangs or crashes . Currently using MacOs 10.14.5 and Postman version 6.7.4

**Comment by saswatds** [comment]
Hello everyone, In the latest version of app 7.3.x we have updated the response viewer stack to be more performant and reduced the memory footprint while showing large responses.

Here is a test sample of me trying to load ~110mb random data
<img width="1004" alt="Screenshot 2019-08-04 at 11 44 13 AM" src="https://user-images.githubusercontent.com/8796447/62420357-6a919000-b6ae-11e9-9734-214b56e9947f.png">

During testing, we found that the amount of data that can be safely shown in the UI greatly depends on the hardware and OS configuration. **So we have set a default maximum renderable response size of 50mb**, which we sanity tested across various devices. 

This is a very pessimistic limit that we have set but you should be able to increase it from your settings. 
<img width="716" alt="Screenshot 2019-08-04 at 11 59 58 AM" src="https://user-images.githubusercontent.com/8796447/62420414-72056900-b6af-11e9-80c5-b31073de53b0.png">
**Note that setting this to really large value will eventually crash the app as the os kills the process on account of consuming a lot of RAM.** 

You can find the latest version of the app here https://www.getpostman.com/downloads/

> We are actively working on a response viewing experience where there will be no limitations on the size of data, but it will take some time in R&D.

Thanks for all your patience.

**Comment by adriennn** [comment]
@saswatds I'm on 7.5.0 and trying to download a 35 MB file from a call to localhost. When the response lands in postman, the app consumes 700 MB, but if I try to save to file, the RAM balloons to 2.5 GB (80% of available memory) then crashes. Is there a memory leak when trying to save the file?
edit: win10 native app

**Comment by HeliosLHD** [comment]
I'm also on 7.5.0 and recently loading 27MB JSON response, one CPU maxes
consumption and memory expands to 2.5GB if the response is set to Pretty,
it seems that it will eventually load after 1-2 minutes.  Performance is
slightly better if I just leave it on Raw.

I'd say things have improved since the original bug in that the program
doesn't spiral and crash, but still not good.

**Comment by balsakokovic** [comment]
Latest Postman 7.10.0 on MacOS Catalina 
JSON response - approx. 33Mb 
App hangs and after a minute goes black and never recovers. It looks like it's stuck Formatting...
<img width="1269" alt="Screenshot 2019-10-31 at 17 29 51" src="https://user-images.githubusercontent.com/25767912/67966956-16db0800-fc05-11e9-9b6a-a0774df67cc5.png">
<img width="1381" alt="Screenshot 2019-10-31 at 17 30 14" src="https://user-images.githubusercontent.com/25767912/67966981-1f334300-fc05-11e9-8960-b7d8e2a7d47b.png">

**Comment by ghost** [comment]
> Latest Postman 7.10.0 on MacOS Catalina
> JSON response - approx. 33Mb
> App hangs and after a minute goes black and never recovers. It looks like it's stuck Formatting...
> <img alt="Screenshot 2019-10-31 at 17 29 51" width="1269" src="https://user-images.githubusercontent.com/25767912/67966956-16db0800-fc05-11e9-9b6a-a0774df67cc5.png">
> <img alt="Screenshot 2019-10-31 at 17 30 14" width="1381" src="https://user-images.githubusercontent.com/25767912/67966981-1f334300-fc05-11e9-8960-b7d8e2a7d47b.png">

Same issue here.

I use PowerShell for the time being to extract the data into a file.

$endpointUri = 'https://domain?param1=2019-09-30&param2=Europe';
$cred = Get-Credential;

Invoke-RestMethod -Uri $endpointUri -Credential $cred -OutFile "D:\filename.json";

**Comment by BaerMitUmlaut** [comment]
Same thing here. ~60mb XML response, when trying to save the file my fans spin up, then black screen, then PostMan seems to go back to idle and never recovers from the black screen. The response also doesnt get saved, which is especially annoying.

**Comment by samuelhaddad-toast** [comment]
Hi @saswatds is there any official word on this? Seems like we are getting a few people experiencing the issue.

**Comment by rickmellor** [comment]
Bump.  I'm running automation via Runner with 50k records and it really struggles.  The UI just hangs and the process is unreliable.

**Comment by saswatds** [comment]
The processing and representation of large user datasets is a performance issue that we are actively looking into.  For getting more out of the existing system we are experimenting a few things internally and also re-architecting some parts of the app. We will keep everyone updated once we have a breakthrough.

**Comment by cksachdev** [comment]
Able to reproduce. 
Postman Version 7.26.1 (7.26.1)
OS: macOS Catalina 10.15.5

Any possible workaround?

**Comment by luigicfh** [comment]
Has this issue been resolved yet?

**Comment by shubhbhargav** [comment]
@lucamaf @cksachdev  May I know which Postman version are you using and which OS? We've pushed across a few fixes 7.18.x along the lines of performance. Ideally it should not have happened beyond that.

If you can share more info about the response size and the memory / CPU behaviour / side-effects, that will be helpful.

**Comment by sprice9161** [comment]
I'm running Postman 10.81. I am have trouble executing a request that performs a file download. Postman crashes (closes) while waiting for the download to occur. I have download size set to "0" and timeout set to "0" in the settings. I think that it fails around 1.2 MB.

I have 2 scenarios:
1 - Single file where a request produces a token that encode and build into the URL for my download endpoint. I then execute the download endpoint.
2 - Multiple files where my request provides a URL that I save, and then place in in its entirety in a request and execute. The downloaded file in this instance will always be a zip file because there are multiple files.

I have created the URL manually, pasted the url into a chrome browser window and successfully downloaded the files. I have tested up to 1 gb.
